src/bo4e/com/verbrauch.py

```diff
@@ -0,0 +1,80 @@
+"""
+Contains Verbrauch and corresponding marshmallow schema for de-/serialization
+"""
+from datetime import datetime
+from decimal import Decimal
+from typing import Protocol
+
+import attr
+from marshmallow import fields, post_load
+from marshmallow_enum import EnumField  # type:ignore[import]
+
+from bo4e.com.com import COM, COMSchema
+from bo4e.enum.mengeneinheit import Mengeneinheit
+from bo4e.enum.wertermittlungsverfahren import Wertermittlungsverfahren
+
+# pylint:disable=line-too-long
+OBIS_PATTERN = r"((1)-((?:[0-5]?[0-9])|(?:6[0-5])):((?:[1-8]|99))\.((?:6|8|9|29))\.([0-9]{1,2})|(7)-((?:[0-5]?[0-9])|(?:6[0-5])):(.{1,2})\.(.{1,2})\.([0-9]{1,2}))"
+
+
+# pylint:disable=too-few-public-methods
+class _StartEndType(Protocol):
+    """
+    an overengineered protocol class
+    """
+
+    startdatum: datetime
+    enddatum: datetime
+
+
+# pylint: disable=unused-argument
+def check_end_is_later_than_start(instance: _StartEndType, attribute, value):
+    """
+    assert that enddatum is later than startdatum
+    """
+    if not instance.enddatum >= instance.startdatum:
+        raise ValueError("enddatum has to be >= startdatum")
```

Comment:
Könnte Sinn machen die beiden Checks in das validators file auszulagern oder? Ich vermute die kommen öfter mal vor.

Reply:
blöderweise jedesmal mit anderen namen. beim auslagern müsste man dem validator dann zwei lambdas mitgeben, die von der instanz auf das von/start bzw. bis/end verweisen

Reply:
Und wenn du es genau so wie es aktuell ist auslagerst? Dann wären zumindest alle die startdatum und enddatum heißen abgedeckt und man könnte es sich analog bauen, wenn es andere Namen gibt, aber dann weiß man schon mal, dass es so was ähnliches schon gibt und kann sich daran orientieren.
Und den regex-check kann man ja vermutlich direkt benutzen, auch bei anderen Namen.